overlord/snapstate/handlers.go

```diff
@@ -522,6 +527,11 @@ func (m *SnapManager) doLinkSnap(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	// Restore configuration for the new revision if available
+	if err = config.RestoreConfigSnapshotMaybe(st, snapsup.Name(), snapsup.Revision()); err != nil {
```

Comment:
We should only do this if this is a revert. On new installs, we want the configuration currently in use to carry over. Can we please have a spread test testing that?  It should fail before this is fixed.

We can use the way we use $SNAP_DATA as a guideline. The same places where we need to copy it over so that the old data goes into a new directory are good places to take a copy of config. After all, the goal is the same: on reverts, bring back the old state.

Do we have any discrepancies there, and if so why?